server/sonar-server/src/main/java/org/sonar/server/startup/ClearRulesOverloadedDebt.java

```diff
@@ -0,0 +1,110 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.startup;
+
+import org.picocontainer.Startable;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.db.DbSession;
+import org.sonar.db.loadedtemplate.LoadedTemplateDto;
+import org.sonar.db.rule.RuleDto;
+import org.sonar.server.db.DbClient;
+
+import static org.sonar.db.loadedtemplate.LoadedTemplateDto.ONE_SHOT_TASK_TYPE;
+
+/**
+ * Clear the overloaded technical debt of rules when SQALE plugin is not installed.
+ *
+ * Should be removed after next LTS
+ *
+ * @since 5.2
+ */
+public class ClearRulesOverloadedDebt implements Startable {
+
+  private static final Logger LOG = Loggers.get(ClearRulesOverloadedDebt.class);
+
+  private static final String TEMPLATE_KEY = "ClearRulesOverloadedDebt";
+
+  private static final String SQALE_LICENSE_PROPERTY = "sonar.sqale.licenseHash.secured";
+
+  private final DbClient dbClient;
+
+  public ClearRulesOverloadedDebt(DbClient dbClient) {
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public void start() {
+    DbSession session = dbClient.openSession(false);
+    try {
+      if (hasAlreadyBeenExecuted(session)) {
+        return;
+      }
+      if (!isSqalePluginInstalled(session)) {
+        clearDebt(session);
+      }
+      registerTasks(session);
+      session.commit();
+    } finally {
+      dbClient.closeSession(session);
+    }
+  }
+
+  private void clearDebt(DbSession session) {
+    int countClearedRules = 0;
+    for (RuleDto rule : dbClient.deprecatedRuleDao().selectAll(session)) {
+      if (isDebtOverridden(rule)) {
+        rule.setSubCharacteristicId(null);
+        rule.setRemediationFunction(null);
+        rule.setRemediationCoefficient(null);
+        rule.setRemediationOffset(null);
+        dbClient.deprecatedRuleDao().update(session, rule);
+        countClearedRules++;
+      }
+    }
+    if (countClearedRules > 0) {
+      LOG.warn("The SQALE model has been cleaned to remove useless data left over by previous migrations. The technical debt of {} rules was reset to their default values.",
+        countClearedRules);
+      LOG.warn("=> As a consequence, the overall technical debt of your projects might slightly evolve during the next analysis.");
+    }
+  }
+
+  private static boolean isDebtOverridden(RuleDto ruleDto) {
+    return ruleDto.getSubCharacteristicId() != null || ruleDto.getRemediationFunction() != null;
+  }
+
+  private boolean isSqalePluginInstalled(DbSession session) {
+    return dbClient.propertiesDao().selectGlobalProperty(session, SQALE_LICENSE_PROPERTY) != null;
+  }
+
+  private boolean hasAlreadyBeenExecuted(DbSession session) {
+    return dbClient.loadedTemplateDao().countByTypeAndKey(ONE_SHOT_TASK_TYPE, TEMPLATE_KEY, session) > 0;
+  }
+
+  private void registerTasks(DbSession session) {
```

Comment:
I know that `loadedTemplate` is badly named, but consistency is more important. I suggest to rename the method `markAsExecuted()` so it does not introduce the term `task` and it's associate with the other method `hasAlreadyBeenExecuted`.